packages/server-wallet/src/models/__test__/objectives.test.ts

```diff
@@ -0,0 +1,62 @@
+import {OpenChannel} from '@statechannels/wallet-core';
+
+import {testKnex as knex} from '../../../jest/knex-setup-teardown';
+import {seedAlicesSigningWallet} from '../../db/seeds/1_signing_wallet_seeds';
+import {Channel} from '../channel';
+import {Objective, ObjectiveChannel} from '../objective';
+
+import {channel} from './fixtures/channel';
+
+const c = channel();
+const objective: OpenChannel = {
+  type: 'OpenChannel',
+  participants: [],
+  data: {
+    targetChannelId: c.channelId,
+    fundingStrategy: 'Direct',
+  },
+};
+beforeEach(async () => {
+  await seedAlicesSigningWallet(knex);
+});
+
+describe('Objective > insert', () => {
+  it.skip('fails to insert / associate an objective when it references a channel that does not exist', async () => {
+    // For some reason this does not catch the error :/
+    await expect(Objective.insert({...objective, status: 'pending'}, knex)).rejects.toThrow();
```

Comment:
@andrewgordstewart do you have any ideas about how to test this properly? It seems that the promise successfully resolves but that an error is thrown afterwards. I'd like to assert that this is the "desired" behaviour (I think?) but I can't seem to catch the error, even when[ wrapping in an arrow function. ](https://jestjs.io/docs/en/expect#tothrowerror)

Reply:
`Objective.insert` is a bit magical. I think it is a different promise than `new Promise()` is.
```suggestion
  it.skip('fails to insert / associate an objective when it references a channel that does not exist', async () => {
    // For some reason this does not catch the error :/
    await expect(
      new Promise(async resolve => { await Objective.insert({...objective, status: 'pending'}, knex); resolve()})
      ).rejects.toThrow();
```

So, something like this might work?

Reply:
<img width="506" alt="Screen Shot 2020-10-21 at 2 05 22 PM" src="https://user-images.githubusercontent.com/8432675/96787443-99c11980-13a6-11eb-8339-e69897863cef.png">

`Nonce.query().insert` returns a `QueryBuilder`, which you _can_ `await`, but it's not a `Promise`.

Reply:
Ok, so `QueryBuilder` is a class that extends `Promise`:
```
export class QueryBuilder<M extends Model, R = M[]> extends Promise<R> { //snip }
```

Perhaps this is incompatible with the implementation `expect(somePromise).rejects`?

Reply:
good thinking; I think I found the issue, though:  I'm awaiting my own `.insert` fn, not the usual one, and it doesn't reject properly when it should.